### PR TITLE
Creating the accumulator state for the eventual malicious validation

### PR DESCRIPTION
**Pull request by benjaminsavage**
Next step along the way towards eventually creating maliciously secure multiplication.

### CODE REVIEW

File: src/protocol/malicious.rs [hunk 1]
```diff
@@ -237,39 +273,38 @@ pub mod tests {
...
+            acc.accumulate_macs(
+                &b_ctx.narrow(&Step::ValidateInput).prss(),
                 RecordId::from(1),
                 b_malicious,
             );
 
             let (ab, rab) = try_join(
                 a_ctx
-                    .narrow(&"SingleMult".to_string())
+                    .narrow("SingleMult")
```
Comment (martinthomson) [inline src/protocol/malicious.rs:289]: Mmm single malt

**Review by martinthomson — APPROVED** [review]
On a tiny screen, so not 100% able to review this. Still this is good stuff. Land it.

File: src/protocol/context.rs [hunk 2]
```diff
@@ -56,6 +70,9 @@ impl<'a, N> ProtocolContext<'a, N> {
             step: self.step.narrow(step),
             prss: self.prss,
             gateway: self.gateway,
+            // TODO: make this work
```
[the remark this answers is not in the document]
Comment (benjaminsavage) [reply src/protocol/context.rs:73]: It complains that it cannot clone this thing... i'm still stuck on it. Need to resolve this for the next diff. I'll work on it there.

File: src/protocol/malicious.rs (hunk 1 above)
Comment (benjaminsavage) [reply src/protocol/malicious.rs:289]: I was thinking the same thing =)

**Pull request merged**